test/Object/ObjectRest_Simple.js

```diff
@@ -0,0 +1,287 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+// Object Rest unit tests
+
+if (this.WScript && this.WScript.LoadScriptFile) { // Check for running in ch
+    this.WScript.LoadScriptFile("..\\UnitTestFramework\\UnitTestFramework.js");
+}
+
+var tests = [
+    {
+        name: "let assignment with simple Object",
+        body: function() {
+            let {a, b, ...rest} = {a: 1, b: 2, c: 3, d: 4};
+            assert.areEqual(1, a);
+            assert.areEqual(2, b);
+            assert.areEqual({c: 3, d: 4}, rest);
+        }
+    },
+    {
+        name: "var assignment with simple Object",
+        body: function() {
+            var {a, b, ...rest} = {a: 1, b: 2, c: 3, d: 4};
+            assert.areEqual(1, a);
+            assert.areEqual(2, b);
+            assert.areEqual({c: 3, d: 4}, rest);
+        }
+    },
+    {
+        name: "Rest in assignment expression",
+        body: function() {
+            ({a, b, ...rest} = {a: 1, b: 2, c: 3, d: 4});

```

Comment:
negative tests like rest should be at the last.

Reply:
what's a negative test?

---
In reply to: [205213734](https://github.com/Microsoft/ChakraCore/pull/5498#discussion_r205213734) [](ancestors = 205213734)

Reply:
what I meant was - invalid syntax.
```let {...rest1, a} ={}```


---
In reply to: [205221255](https://github.com/Microsoft/ChakraCore/pull/5498#discussion_r205221255) [](ancestors = 205221255,205213734)